aws_common/src/sdk_utils/client_configuration_provider.cpp

```diff
@@ -26,6 +26,24 @@
 namespace Aws {
 namespace Client {
 
+/**
+ *
+ */
+class NoRetryStrategy : public DefaultRetryStrategy
```

Comment:
@dabonnie Should this be inheriting from `RetryStrategy` instead of `DefaultRetryStrategy`?